src/sinks/gcp/stackdriver_metrics.rs

```diff
@@ -0,0 +1,216 @@
+
+        let end_time = metric.data.timestamp.unwrap_or_else(chrono::Utc::now);
+
+        let point_value = match metric.data.value {
+            MetricValue::Counter { value } => {
+                metric_kind = "CUMULATIVE";
+                interval.insert(
+                    "startTime",
+                    self.started.to_rfc3339_opts(SecondsFormat::Nanos, true),
+                );
+                interval.insert(
+                    "endTime",
+                    end_time.to_rfc3339_opts(SecondsFormat::Nanos, true),
+                );
+                value
+            }
+            MetricValue::Gauge { value } => {
+                metric_kind = "GAUGE";
+                interval.insert(
+                    "endTime",
+                    end_time.to_rfc3339_opts(SecondsFormat::Nanos, true),
+                );
+                value
+            }
+            not_supported => {
+                warn!("Unsupported metric kind: {:?}.", not_supported);
+                return None;
+            }
+        };
+
+        let series = serde_json::json!({
+            "metric": { "type": metric_type, "labels": metric_labels },
+            "resource": {
+                "type": self.config.resource.tpe.clone(),
+                "labels": self.config.resource.labels.clone()
+            },
+            "metricKind": metric_kind,
+            "valueType": "INT64",
+            "points": [ serde_json::json!({
+                "interval": interval,
+                "value": serde_json::json!({
+                    "int64Value": (point_value as i64).to_string()
+                })
+            })]
+        });
+
+        Some(series)
+    }
+
+    async fn build_request(
+        &self,
+        time_series: Self::Output,
+    ) -> crate::Result<hyper::Request<Vec<u8>>> {
+        let time_series = serde_json::json!({ "timeSeries": time_series });
+
+        let body = serde_json::to_vec(&time_series).unwrap();
+        let uri: Uri = format!(
+            "https://monitoring.googleapis.com/v3/projects/{}/timeSeries",
+            self.config.project_id
+        )
+        .as_str()
+        .parse()
+        .unwrap();
+
+        let mut request = hyper::Request::post(uri)
+            .header("content-type", "application/json")
+            .body(body)
+            .unwrap();
+
+        request.headers_mut().insert(
+            AUTHORIZATION,
+            self.token
+                .header_value()
+                .unwrap()
+                .as_str()
+                .parse::<HeaderValue>()
+                .unwrap(),
+        );
+
+        Ok(request)
+    }
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        rate_limit_num: Some(1000),
+        rate_limit_duration_secs: Some(1),
+        ..Default::default()
+    };
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "gcp_stackdriver_metrics")]
+impl SinkConfig for StackdriverConfig {
```

Comment:
I think we would prefer seeing these nearer the top of the file, just for consistency with other sinks.